test/failure/bad-hcom-step.prl

```diff
@@ -1,10 +0,0 @@
-Thm BoolSym : [
-  (-> 
-   [a b : bool]
-   (path {_} bool a b)
-   (path {_} bool b a))
-] by [
-  { lam a. lam b. lam p. <i> 
-     `(hcom{0 ~> 1} bool ,a [i=0 {j} (@ ,p j)] [i=1 {_} ,a])};
-  [auto-step; head-expand]
```

Comment:
Previously, this test should have failed for two reasons:

1. The overly conservative `safeStep` stuff would have thrown an error when trying to reduce the `hcom`.
2. After the machine was fixed, this test continued to fail for a different reason: head expansion succeeded, leaving behind a still-unsolved goal. This describes the period of time before the `<i>` and `lam` development tactics automatically called `auto` on their auxiliary subgoals.
3. Finally, after I changed the development tactics to call `auto`, this test would have failed because an impossible subgoal was generated by `auto`, which is now fixed in this PR.

A very complicated story!